### PR TITLE
Avoid some redirects in docs

### PR DESCRIPTION
**Pull request by dreis2211**
Hi,

this PR polishes some links to avoid redirects (and one broken anchor).

Cheers,
Christoph

### CODE REVIEW

File: spring-boot-project/spring-boot-docs/src/docs/asciidoc/getting-started.adoc [hunk 1]
```diff
@@ -346,7 +346,7 @@ If you need to solve a specific problem, check there first.
 
 You can shortcut the steps below by going to https://start.spring.io and choosing the "Web" starter from the dependencies searcher.
 Doing so generates a new project structure so that you can <<getting-started-first-application-code,start coding right away>>.
-Check the {spring-initializr-docs}/#user-guide[Spring Initializr documentation] for more details.
+Check the {spring-initializr-docs}[Spring Initializr documentation] for more details.
```
Comment (dreis2211) [inline spring-boot-project/spring-boot-docs/src/docs/asciidoc/getting-started.adoc:349]: I wasn't sure what more details the link to Initializr would give in that context, but I didn't want to remove it just yet. Just let me know and I'll remove it if you think it's not providing any benefit here.

Comment (snicoll) [reply spring-boot-project/spring-boot-docs/src/docs/asciidoc/getting-started.adoc:349]: Yeah it's a good point. I wonder if https://github.com/spring-io/start.spring.io/blob/main/USING.adoc wouldn't be a better link now. What do you think?

Comment (dreis2211) [reply spring-boot-project/spring-boot-docs/src/docs/asciidoc/getting-started.adoc:349]: Seems like a better link - I'll adjust my PR.

**Comment by snicoll** [comment]
Thank you very much again, Christoph.

**Pull request closed without merging**